graphql/types.go

```diff
@@ -43,6 +43,18 @@ func (e *Enum) enumValues() []string {
 	return e.Values
 }
 
+// CustomScalar is a leaf value with a custom Unwrapper function.
+type CustomScalar struct {
+	TypeName  string
+	Unwrapper func(interface{}) (interface{}, error)
```

Comment:
should we make this part of `Scalar`?

how does introspection handle this? can we add an introspection test case?

Reply:
It was already a part of the introspection, but I think I agree that it would be nicer/safer to not have to change all those locations.